colexec: implement Close in Columnarizer

asubiotto opened the pull request:
A Columnarizer needs to call ConsumerClosed on the wrapped input tree since
those inputs may not be drained in some cases.

Release note: None

Fixes #51179

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51376)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
Oh yeah, this makes sense.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51376#-:-MC7jvYE3wNjjPRSyxDt:bb7rgb8)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/colexec/columnarizer.go, line 46 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51376#-MC7jSKf1JRTTBrhgZjb:-MC7jSKf1JRTTBrhgZjc:b-hifw9s) ([raw file](https://github.com/cockroachdb/cockroach/blob/9083d5651fd83fd9de7e8070202f3d0adfb0663d/pkg/sql/colexec/columnarizer.go#L46)):*
> ```Go
> }
> 
> var _ colexecbase.Operator = &Columnarizer{}
> ```

nit: we should add `var _ Closer` assertion here.
___
*[pkg/sql/colexec/columnarizer.go, line 103 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51376#-MC7jYDBC5mLl787bc4V:-MC7jYDBC5mLl787bc4W:b1gzqcc) ([raw file](https://github.com/cockroachdb/cockroach/blob/9083d5651fd83fd9de7e8070202f3d0adfb0663d/pkg/sql/colexec/columnarizer.go#L103)):*
> ```Go
> 	for ; nRows < coldata.BatchSize(); nRows++ {
> 		row, meta := c.input.Next()
> 		if meta != nil {
> ```

Somewhat related: do you think we should check `meta.Err` here?


<!-- Sent from Reviewable.io -->

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51376)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/sql/colexec/columnarizer.go, line 103 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51376#-MC7jYDBC5mLl787bc4V:-MC7klek6IXpiR1yPhz7:bba0yfu) ([raw file](https://github.com/cockroachdb/cockroach/blob/9083d5651fd83fd9de7e8070202f3d0adfb0663d/pkg/sql/colexec/columnarizer.go#L103)):*
<details><summary><i>Previously, yuzefovich wrote…</i></summary><blockquote>

Somewhat related: do you think we should check `meta.Err` here?
</blockquote></details>

Yeah, but I'm doing that in #51375 with the test.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51376#-:-MC7lv_y8VD-UuRaiEOT:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/colexec/columnarizer.go, line 103 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51376#-MC7jYDBC5mLl787bc4V:-MC7ljED9m-oo5li7d1T:byvd0kc) ([raw file](https://github.com/cockroachdb/cockroach/blob/9083d5651fd83fd9de7e8070202f3d0adfb0663d/pkg/sql/colexec/columnarizer.go#L103)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

Yeah, but I'm doing that in #51375 with the test.
</blockquote></details>

Oh, ok, I haven't looked at that PR yet.


<!-- Sent from Reviewable.io -->

Commit:
colexec: implement Close in Columnarizer

A Columnarizer needs to call ConsumerClosed on the wrapped input tree since
those inputs may not be drained in some cases.

Release note: None

asubiotto commented:
fixed lint error

bors r=yuzefovich
